flocker/testtools/_base.py

```diff
@@ -104,6 +128,112 @@ def mktemp(self):
         return make_temporary_directory(self).child('temp').path
 
 
+def _fix_twisted_logs(log_content):
+    """
+    Split the Eliot logs out of a Twisted log.
+
+    :param Content log_content: A text content object that contains a Twisted
+        log.
+    :return: The log split into two, the first containing the core Twisted log
+        messages and the second containing line-separated Eliot JSON messages.
+    :rtype: (Content, Content)
+    """
+    twisted_lines, eliot_lines = tee(_iter_content_lines(log_content))
+    return (
+        Content(UTF8_TEXT, partial(_pure_twisted_logs, twisted_lines)),
+        Content(
+            UTF8_TEXT,
+            lambda: _prettyformat_lines(_iter_eliot_logs(eliot_lines)),
+        ),
+    )
+
+
+def _pure_twisted_logs(log_lines):
+    """
+    Given an iterable of lines of bytes of Twisted logs, yield the lines that
+    are not Eliot data.
+    """
+    return (line for line in log_lines if _get_eliot_data(line) is None)
```

Comment:
:question: It probably doesn't matter, since this code isn't performance sensitive, but it looks like `_get_eliot_data` is being called twice on every line.

You could perhaps do something like
```
lines_with_eliot_data = (line, get_eliot_data(line)) for line in _iter_content_lines(log_content)
t, e = tee(lines_with_eliot_data)
twisted_lines = line for (line, eliot_data) in t if eliot_data is None
eliot_lines = eliot_datat for (line, eliot_data in e if eliot_data is not None
```
but I'm not sure if the additional complication is worth it.